nexus/tests/integration_tests/ip_pools.rs

```diff
@@ -275,6 +275,32 @@ async fn test_ip_pool_list_dedupe(cptestctx: &ControlPlaneTestContext) {
     let silo3_pools = pools_for_silo(client, "silo3").await;
     assert_eq!(silo3_pools.len(), 1);
     assert_eq!(silo3_pools[0].identity.name, "pool1");
+
+    // this is a great spot to check that deleting a pool cleans up the links!
+
+    // first we have to delete the range, otherwise delete will fail
+    let url = "/v1/system/ip-pools/pool1/ranges/remove";
+    NexusRequest::new(
+        RequestBuilder::new(client, Method::POST, url)
+            .body(Some(&range1))
+            .expect_status(Some(StatusCode::NO_CONTENT)),
+    )
+    .authn_as(AuthnMode::PrivilegedUser)
+    .execute()
+    .await
+    .expect("Failed to delete IP range from a pool");
+
+    object_delete(client, "/v1/system/ip-pools/pool1").await;
+
+    let silo1_pools = pools_for_silo(client, "silo1").await;
+    assert_eq!(silo1_pools.len(), 1);
+    assert_eq!(silo1_pools[0].id(), pool2.id());
+
+    let silo2_pools = pools_for_silo(client, "silo2").await;
+    assert_eq!(silo2_pools.len(), 0);
+
+    let silo3_pools = pools_for_silo(client, "silo3").await;
+    assert_eq!(silo3_pools.len(), 0);
```

Comment:
This worth doing but it's not as good a test as I hoped for _this_ change, because this query actually joins ip_pool and ip_pool_resource when listing links, so if the pool has been deleted, the links won't show up even if they're in the table. However, it does confirm that you can delete the pool with links outstanding.